provider/ec2/instance_information.go

```diff
@@ -0,0 +1,30 @@
+package ec2
+
+import (
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/constraints"
+	"github.com/juju/juju/environs"
+	"github.com/juju/juju/environs/instances"
+)
+
+var _ environs.InstanceTypesFetcher = (*environ)(nil)
+
+// InstanceTypes implements InstanceTypesFetcher
+func (e *environ) InstanceTypes(c constraints.Value) (instances.InstanceTypesWithCostMetadata, error) {
+	iTypes, err := e.supportedInstanceTypes()
+	if err != nil {
+		return instances.InstanceTypesWithCostMetadata{}, errors.Trace(err)
+	}
+	for i, t := range iTypes {
+		iTypes[i].Cost = t.Cost / 1000
```

Comment:
Nope. I must have been unclear, sorry.

What I meant was to include a divisor in the result struct, not to truncate/drop precision in the cost field. i.e. the results should include:

 - the cost (e.g. in deci-cents)
 - a divisor (e.g. of 1000) to get to dollars, or whatever the unit is
 - the currency/unit (e.g. $USD/h)

e.g. with cost=1234, divisor=1000, currency/unit=$USD/h, the GUI could display "1.234 $USD/h".